### ISSUE TITLE
Failed training jobs save many wandb media files

### ISSUE THREAD

**Issue by oliverwm1** [ai2cm/ace#370]
ai2cm/ace#164 is cropping up again. Maybe because we're using a more recent wandb version and the file paths/names of these media files changed.

For example: https://beaker.org/ds/01J0PB4FF446GQCVWXR9DG9XAF/details

**Comment by oliverwm1** [comment]
@brianhenn pointed out that this is likely happening because the linked run crashed. The wandb clean up only happens if the run makes it successfully through the last epoch. A lot of my runs these days do either crash or are manually cancelled early, so it might be worth revisting this, but I'm not sure how to implement without potentially messing up the wandb logging.

**Comment by brianhenn** [comment]
Related -- beaker is having trouble sometimes with the amount of files that are generated in these temporary directories: https://github.com/allenai/beaker-compute/issues/165 . It is trying to sync these files to results that are later deleted, which is unnecessary and also apparently slowing down the nodes.

**Comment by brianhenn** [comment]
The suggestion from Michal G was to change the wandb temporary directory to be outside of beaker experiment results directory, e.g., `/tmp`. I am not sure if this would cause beaker training jobs to not resume correctly in terms of wandb logging though.

**Comment by oliverwm1** [comment]
> The suggestion from Michal G was to change the wandb temporary directory to be outside of beaker experiment results directory, e.g., /tmp. I am not sure if this would cause beaker training jobs to not resume correctly in terms of wandb logging though.

Yeah, we rely on the wandb directory to be in the "results" directory to allow resuming. Last I checked there was no way to work around having the "resume" information in the same place as all the media. But maybe there has been some movement from wandb on this—if I recall correctly, there was a wandb GH issue about it.

**Comment by brianhenn** [comment]
One thought is to ditch the wandb resuming dependence on the local wandb directory. This could be accomplished as long as we have a way of knowing the wandb id of the preempted job. It might be possible to get this out of beaker.

**Comment by oliverwm1** [comment]
Does wandb have a way to resume runs if you know the prior ID? I thought it relied on the wandb dir being in the same place, and I'm hesitant to roll our own too much on implementing the resume capability. But if it would be straightforward, I'm open to other options.

**Comment by brianhenn** [comment]
I think so: https://docs.wandb.ai/guides/runs/resuming/#resume-a-run-that-must-use-the-same-run-id. Main weakness is being able to get the ID robustly from the beaker logs.

Presumably that'd also mean that anything that wandb has locally saved and hasn't yet logged/synced is lost, but that's the same state we're in with our current resumption approach I think.

**Comment by oliverwm1** [comment]
Right, but we're relying on idempotency here for this to work easily with beaker preemption. I.e. the exact same call can be made when first initializing the run and whenever it is restarting. Sounds like things would get a bit hacky if we have to check for possible prior beaker runs and get the wandb ID (e.g. now the restarting wouldn't work on other compute systems).

**Comment by brianhenn** [comment]
True, but something like this pretty much follows our current beaker/perlmutter resumption approach I think?
```
wandb_id_file = os.path.join(config.experiment_dir, 'wandb_id')

if os.path.exists(wandb_id_file):
    wandb_id = read_wandb_id_file(wandb_id_file)
    current_args['id'] = wandb_id
    # Initialize the wandb run
    wandb.init(**current_args)
else:
     # Initialize the wandb run
    wandb.init(**current_args)
    # Get the run ID
    wandb_id = wandb.run.id
    with open(wandb_id_file, 'w') as f:
        f.write(wandb_id)
```

**Comment by oliverwm1** [comment]
Okay, if something simple similar to that works, I'd be down!
